### PR TITLE
[DTF-3234] Deep dive blog post

### PR DESCRIPTION
**Pull request by alexmorley**

### CODE REVIEW

File: docs/deep-dive.md [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+
+
+<img width="437" alt="period-address-lit" src="https://user-images.githubusercontent.com/13655521/231153786-caf0d3a3-02c0-4b18-9361-6f536a8f18ba.png">
+
+
+We should be able to do a bit better here, and have the IDE/compile explicitly tell us that we’re missing a witness for the underlying type of FHIRDateTime \/ Period but that is sadly something we haven’t figured out how to render at present - even the compiler error is Cannot prove that com.babylonhealth.lit.core.model.Address is a viable type for union com.babylonhealth.lit.hl7.UnionAliases.Union\_0934386166, whereas ideally we would want that union type to be dealiased before being displayed. Nonetheless, the essential information that *something is wrong* is being propagated directly to the developer as they work, and the underlying required type is discoverable with a few clicks through the source code.
```
Comment (aelred) [inline docs/deep-dive.md:152]:
```suggestion
We should be able to do a bit better here, and have the IDE/compiler explicitly tell us that we’re missing a witness for the underlying type of FHIRDateTime \/ Period but that is sadly something we haven’t figured out how to render at present - even the compiler error is Cannot prove that com.babylonhealth.lit.core.model.Address is a viable type for union com.babylonhealth.lit.hl7.UnionAliases.Union\_0934386166, whereas ideally we would want that union type to be dealiased before being displayed. Nonetheless, the essential information that *something is wrong* is being propagated directly to the developer as they work, and the underlying required type is discoverable with a few clicks through the source code.
```

File: docs/deep-dive.md [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+
+Extract all Coding objects from a resource
+
+```scala
+def extractAllCodings[T <: Resource](r: T): LitSeq[Coding] = r.nodalGetByClass(classOf[Coding])
+```
+
+Anyway of getting some numbers here? In comparison to HAPI for example?
+
+Could link to the CI performance regression tests here
```
Comment (aelred) [inline docs/deep-dive.md:225]: Looks like a TODO

Comment (alexmorley) [inline docs/deep-dive.md:225]:
```suggestion
```

File: docs/deep-dive.md [hunk 3]
```diff
@@ -0,0 +1,265 @@
+## Lit: Yet another FHIR library
+
+Want an intro? Check out [the article](https://medium.com/@alexmorley/lit-yet-another-fhir-library-dfe663b36c14) on Medium.
+
+**What does Lit look like?**
+
+Here is a function to construct a FHIR BMI object, given a timestamp at which the observation was take, a double (representing the measured BMI) and a patientId (for the patient to which the observation applies):
```
Comment (DrGreggles) [inline docs/deep-dive.md:7]:
```suggestion
Here is a function to construct a FHIR BMI object, given a timestamp at which the observation was taken, a double (representing the measured BMI) and a patientId (for the patient to which the observation applies):
```

**Pull request merged**